File: Python/374.Guess Number Higher or Lower.py
#猜数游戏，在1到n间猜一个数，每次猜错告诉是猜大了还是小了，直至猜对。
#-1,猜大了。1，猜小了。0，猜对了。
class Solution(object):
    def guessNumber(self, n):
        l , r = 1 , n
        while l<=r:
            mid = (l+r)//2
            response = self.guess(mid)
            if response==0:
                return mid
            elif response==1:
                l = mid+1
            else:
                r = mid-1

    def guess(self,x):
        if x==6:    return 0
        elif x>6:   return -1
        else:       return 1

s = Solution()
print(s.guessNumber(10))

File: C++/Binary Search/374.Guess Number Higher or Lower.cpp
/*
猜测数的大小。数为从1到n，猜测别人挑选的是哪个。
每次调用API函数guess，告诉挑选的数与答案的大小比较。
-1表示别人的数小。1表示别人的数大。0表示猜对了。
*/
/*
思路：二分法。
 */

#include <iostream>
using namespace std;

class Solution {
public:
	int guessNumber(int n) {
        int l = 1, r = n;
        while(l <= r)
        {
        	int mid = l + (r - l)/2;
        	int res = guess(mid);
        	if(res == 0)	return mid;
        	else if(res==1)	l = mid+1;
        	else r = mid-1;
        }
        return -1;
    }
};

int main()
{
	Solution s;
	cout<<s.guessNumber(256)<<endl;
}

